### PR TITLE
app: args flags to maxNp and strategy

### PR DESCRIPTION
**Pull request by angelopra**

### CODE REVIEW

File: src/App.java [hunk 1]
```diff
@@ -1,16 +1,61 @@
...
 
 public class App {
     public static void main(String[] args) throws Exception {
-        
-        //First arg is the max number of processes in the ready queue
-        if(args.length != 1) {
-            System.out.println("Usage: java App <max number of processes in the ready queue>");
-            return;
+        int npIndex = Arrays.asList(args).indexOf("-np"); // index of the number of process flag in args array
+        int stratIndex = Arrays.asList(args).indexOf("-strat"); // index of the strategy flag in args array (rr or sjf)
```
Comment (gabrielbergoc) [inline src/App.java]: Why not make only one conversion from array to list?

File: src/App.java [hunk 2]
```diff
@@ -1,16 +1,61 @@
...
+        int maxProcessReadySize = 100; // default max number of processes in the ready queue
+        String stratName = "sjf"; // default strategy: SJF
+        SchedulingStrategy strategy;
+
+        if (npIndex != -1) { // the -np flag was used
+            try {
+                maxProcessReadySize = Integer.parseInt(args[npIndex+1]);
+                if (maxProcessReadySize <= 0) {
+                    throw new NumberFormatException();
```
Comment (gabrielbergoc) [inline src/App.java:21]: Maybe add a message to the user?

File: src/App.java [hunk 3]
```diff
@@ -1,16 +1,61 @@
...
+                }
+            } catch (Exception e) {
+                if (e instanceof ArrayIndexOutOfBoundsException) { // if the user uses the flag and don't inform an value in sequence
+                    System.err.println("If using -np flag, inform an value (>0)");
```
Comment (gabrielbergoc) [inline src/App.java]: Typo (_a_ value)

File: src/App.java [hunk 4]
```diff
@@ -1,16 +1,61 @@
...
+                    return;
+                }
+                if (e instanceof NumberFormatException) { // if the value informed wasn't valid (couldn't convert to int, or <=0)
+                    System.err.println("Inform an valid value for np (>0)");
+                    return;
+                }
```
Comment (gabrielbergoc) [inline src/App.java]: Maybe it's best to put this message in the exception constructor and don't catch it.

File: src/App.java (hunk 3 above)
Comment (gabrielbergoc) [inline src/App.java]: Maybe rethrow with this message?

File: src/App.java [hunk 5]
```diff
@@ -1,16 +1,61 @@
...
+                    throw new IllegalArgumentException();
+                }
+            } catch (Exception e) {
+                if (e instanceof ArrayIndexOutOfBoundsException) { // if the user uses the flag and don't inform an value in sequence
+                    System.err.println("If using -strat flag, inform an value ('rr' or 'sjf')");
+                    return;
+                }
+                if (e instanceof IllegalArgumentException) { // if the value informed wasn't valid (not 'sjf' nor 'rr')
+                    System.err.println("Inform an valid strategy ('rr' or 'sjf')");
+                    return;
```
Comment (gabrielbergoc) [inline src/App.java]: Same things here.

File: src/App.java [hunk 6]
```diff
@@ -1,16 +1,61 @@
...
+                }
+            }
+        }
+
+        if (stratName.equals("rr")) {
+            strategy = new RoundRobin();
+        } else {
+            strategy = new ShortestJobFirst();
```
Comment (gabrielbergoc) [inline src/App.java:54]: What if `stratName.equals('HEE HEE')`?

Comment (gabrielbergoc) [reply src/App.java:54]: Suggestion: use a map

**Review by gabrielbergoc — CHANGES_REQUESTED** [review]

File: src/App.java (hunk 2 above)
Comment (angelopra) [reply src/App.java:21]: The message is given in the catch block

File: src/App.java (hunk 4 above)
Comment (angelopra) [reply src/App.java]: I use the same exception as the parseInt so I can catch both and display the same message. If I put it in the constructor, the message will not appear if the user types a value that cannot be parsed to int.

File: src/App.java (hunk 6 above)
Comment (angelopra) [reply src/App.java:54]: > What if `stratName.equals('HEE HEE')`?

Won't be, it is treated before. The else is 'sjf'.

Comment (angelopra) [reply src/App.java:55]: > 

If I use a Map, one instance of a strategy will be for useless.

Comment (angelopra) [reply src/App.java:54]: > Suggestion: use a map

If I use a Map, one instance of a strategy will be useless.

**Review by gabrielbergoc — APPROVED** [review]

**Pull request merged**